File: solutions/2015/1_2.py
displacement = {"(": 1, ")": -1}


def process(buff: str) -> None:
    floor = 0
    for i, move in enumerate(buff, 1):
        floor += displacement[move]
        if floor == -1:
            break
    print(i)


def solution() -> None:
    pass

File: solutions/2015/1_2.rs
fn move_until_basement(buff: &str) -> usize {
    let mut floor: i32 = 0;
    for (i, ch) in buff.chars().enumerate() {
        if ch == '(' {
            floor += 1;
        } else if ch == ')' {
            floor -= 1;
        }
        if floor == -1 {
            return i + 1;
        }
    }
    return 0;
}

pub fn process(buff: &str) {
    println!("{}", move_until_basement(&buff));
}

pub fn solution() {}
